auth/auth.go

```diff
@@ -186,6 +234,12 @@ func (auth *Authenticator) getPrincipal(docID string, factory func() Principal)
 			}
 			if channelsChanged {
 				changed = true
+				if user, ok := princ.(User); ok {
+					err = auth.InheritedChannels(user, princ)
```

Comment:
I think we end up doing most of this work inside `rebuildChannels` anyway (the exception being the role channels).

I'm wondering if it's worth putting the warning/limit logic inside there (behind a parameter to limit to only this codepath?) to avoid having to iterate the full set of channels twice.